text/0000-pack-publish-buildpack.md

```diff
@@ -42,6 +42,102 @@ There are four mechanisms for publishing a buildpack:
 * Direct, which commits directly to the local Git repository for the index and makes a Git push to the origin
 * Yank, which removes a buildpack from the registry
 
+## Configuration
+
+##### Base Schema
+```toml
+[registries]
+default="<name>"
+
+[registries.<name>]
+type="<registry type>"
+# <additional per "type" config entries>
+```
+
+##### Type: GitHub Schema
+```toml
+[registries.<name>]
+type="github"
+url="<url to git repo>"
+issues-url="<url to issues for publishing>" # optional, default: "<url>/issues"
+```
+
+##### Type: GitHub Schema
+```toml
+[registries.<name>]
+type="git"
+url="<path to git repo>"
+```
+
+
+#### Default
+
+The following could be an implied default when no registries are present
+
+```
+[registries]
+default="official"
+
+[registries.official]
+type="github"
+url="https://github.com/buildpack/registry"
+
+[registries.official.github]
+issues-url="https://github.com/buildpack/registry/issues"
+```
+
+## Format
+
+### Type: Git
+
+A commit would be created with the following message:
+
+```
+[<ADD|YANK>] <namespace>/<name>@<version>
+
+<user message (optional)>
+```
+
+##### Example
+
+```
+[YANK] buildpacks-io-samples/hello-universe@0.0.1
+
+Major bug issues found. See https://github.com/buildpacks/samples/issues.
+```
+
+### Type: GitHub issue
+
+An issue would be created with the following content:
+
+- Title: `[<ADD|YANK>] <namespace>/<name>@<version>`
+- Body:
+
+        <user message (optional)>
+
+        ### Diff
+
+        ```patch
+        <patch contents>
```

Comment:
i'm not sure if i want to put the patch contents in the diff. It will be harder to validate than just some raw json (which can then be converted into the proper commit without the risk of the diff containing anything "bad")

Reply:
"bad" like an invalid operation or ill formatted?

The main reasoning for this approach was to minimize the logic layer for applying the change. In this scenario the logic of applying is done at the client (`pack`) versus any automation that would need to be replicated elsewhere you wanted to host a registry. 

Given that the initial goal was to strive for something where the logic mostly lived on the client. The set of responsibilities could have looked like this:

client responsibilities:
- validate operation
- create operation diff
- sign diff
- create issue

registry responsibilities:
- check integrity
- check permissions
- apply diff
- commit

Thinking about it further there is still a big risk of ill behaved clients (or manual alterations) that would yield in a bad operation. A more secure approach, and maybe something you were already thinking about, would be to have the client create an operation (as per your suggestion JSON) and add a mechanism to `pack` to read this format and apply it via the "git/native" mechanism.

client responsibilities:
- validate operation
- create operation payload
- create issue

registry responsibilities:
- check permissions
- use something like `pack publish-buildpack --native --input <file>` to apply
    - validate the operation (again)
    - make changes
    - commit
    - push

Reply:
I don't think we can trust the diff--it could be malicious or malformed--so we'd need to have logic to process it anyways. It also makes it harder to manually submit the PR (which we may need in the future if we run a service).

I'd propose a similar structure like:

- Title: `[ADD/YANK] <ns>/<name>@<version>`
- Body:

      <user message (optional)>

      ### Data
      
      id = <ns>/<name>
      version = <version>
      addr = <address-to-image>
      digest = <digest-of-image>